crates/bevy_reflect/src/path/mod.rs

```diff
@@ -478,6 +478,13 @@ impl<const N: usize> From<[Access<'static>; N]> for ParsedPath {
     }
 }
 
+impl<'a> TryFrom<&'a str> for ParsedPath {
```

Comment:
Any reason not to also implement `FromStr`?

Reply:
In the issue and after a bit of reading it seems TryFrom<&str> does replaces FromStr.

I can add FromStr if people want

Reply:
I've heard arguments that it is semantically different (more for parsing strings) than `TryFrom`.

While we `TryFrom` is probably good enough (if not better, as you mentioned), I feel we might as well implement both.

Thoughts?

Reply:
Just looked for some more arguments for FromStr and aside from semantic differences it is very nice that it enables `”literal”.parse()`.

I’ll add FromStr as well

Reply:
@MrGVSV Actually... I think I found the practical reason for why `TryFrom` is more flexible. As far as I can tell it is not possible to implement `FromStr` in this case because the associated error type requires a lifetime parameter (due to referencing the input &str) and the trait definition does not accept one.

https://stackoverflow.com/questions/28931515/how-do-i-implement-fromstr-with-a-concrete-lifetime
https://www.reddit.com/r/rust/comments/7syu6z/where_do_you_declare_a_lifetime_for_fromstrerr/

Full disclosure I am pretty new to rust and lifetimes in particular are a difficult concept for me, so I could be absolutely wrong here lol.

Reply:
Ah good point. I suppose we could use a `'static` lifetime for the error, but that makes it really easy for an unknowing user to accidentally perform unnecessary allocations when using this API.

We can leave it out for now then until someone decides they really need it 🙂

Reply:
Yeah best I could do was something like this which is obviously a leak lol
```
impl FromStr for ParsedPath {
    type Err = ReflectPathError<'static>;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let static_s: &'static str = Box::leak(s.to_string().into_boxed_str());
        ParsedPath::parse(static_s)
    }
}
```
Probably best to leave it out like you said